### ISSUE TITLE
Selective hydration

### ISSUE THREAD

**Issue by KoderFPV** [theKashey/react-imported-component#236]
Hi,

I would like to know if selective hydration can be achieved with this library.

As far I understand rehydrateMarks forces the app to wait until all js are downloaded before it starts the hydration process.

So we have to get everything to have anything working. 
In selective hydration react can prior what to download and hydrate first based on user interaction.

am I right?

**Comment by theKashey** [comment]
Selective Hydration might not be what you think it is. Apparently, if you will try to operate with your application when it's not fully loaded there is a chance that all "not yet loaded" boundaries will fall into suspense mode. Any context update "above" will cause it.

So you dont need to wait for `rehydrateMarks`, but you _might_ want to do that.

**Comment by KoderFPV** [comment]
> Selective Hydration might not be what you think it is.

Yes, I am still learning all the new R18/19 features, and thay're quite confusing sometimes. 

However, I achieved something interesting in nextjs, and now I would like to do the same in React without a framework.
But, I am not sure if waiting for marks allows me that, or even if it's related to this library

In nextjs, a I created dev toolbar, it is an asynchronous server component with promise inside of it. This promise resolves after 5000ms (because it's waiting for some data)

And In my understanding, I was able to, load the whole page, make it interactive, before whole page was fully loaded.

Is it something that react-imported-component could help with?

[screen-capture (2).webm](https://github.com/theKashey/react-imported-component/assets/7673542/d6b83736-b79c-4e5a-b9a6-a7d7ab3cd6cd)

![image](https://github.com/theKashey/react-imported-component/assets/7673542/7a24c0e9-bb37-4524-9831-ccbbf73422e4)

![image](https://github.com/theKashey/react-imported-component/assets/7673542/97f11ed8-7cff-41f5-a065-2f28bf93f8a8)
